transport/src/main/java/io/netty/bootstrap/ChannelInitializerExtension.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.bootstrap;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ServerChannel;
+
+/**
+ * A channel initializer extension make it possible to enforce rules and apply modifications across multiple,
+ * disconnected uses of Netty within the same JVM process.
+ * <p>
+ * For instance, application-level firewall rules can be injected into all uses of Netty within an application,
+ * without making changes to such uses that are otherwise outside the purview of the application code,
+ * such as 3rd-party libraries.
+ * <p>
+ * Channel initializer extensions are <em>not</em> enabled by default, because of their power to influence Netty
+ * pipelines across libraries, frameworks, and use-cases.
+ * Extensions must be explicitly enabled by setting the {@link #EXTENSIONS_SYSTEM_PROPERTY} to {@code serviceload}.
```

Comment:
1. Consider `{@link #EXTENSIONS_SYSTEM_PROPERTY}` -> `{@value #EXTENSIONS_SYSTEM_PROPERTY}`
2. Consider documenting `log` variant

Reply:
The log variant is documented on the constant.